### PR TITLE
fix:Added tagged_state_was_set_grows and state_was_set_grows to the state library

### PR DESCRIPTION
**Pull request by fabian-hk**

### CODE REVIEW

File: src/lib/state/DY.Lib.State.Typed.fst [hunk 1]
```diff
@@ -156,6 +156,18 @@ val state_was_set:
...
+  #a:Type -> {|ev:local_state a|} ->
+  tr1:trace -> tr2:trace ->
+  prin:principal -> sid:state_id -> e:a  ->
+  Lemma
+  (requires state_was_set tr1 prin sid e /\ tr1 <$ tr2)
+  (ensures state_was_set tr2 prin sid e)
+  [SMTPat (state_was_set tr1 prin sid e); SMTPat (tr1 <$ tr2)]
+let state_was_set_grows #a #ev tr1 tr2 prin sid e =
+  reveal_opaque (`%state_was_set) (state_was_set #a);
+  ()
```
Comment (qaphla) [inline src/lib/state/DY.Lib.State.Typed.fst:169]: Nit: I think it's a bit cleaner to remove the `; ()` from the end, though it does arguably make diffs nicer if we need to add more lines in to the proof. Likewise in `tagged_state_was_set_grows`.

File: src/lib/state/DY.Lib.State.Typed.fst [hunk 2]
```diff
@@ -156,6 +156,17 @@ val state_was_set:
...
+val state_was_set_grows:
+  #a:Type -> {|ev:local_state a|} ->
+  tr1:trace -> tr2:trace ->
+  prin:principal -> sid:state_id -> e:a  ->
+  Lemma
+  (requires state_was_set tr1 prin sid e /\ tr1 <$ tr2)
+  (ensures state_was_set tr2 prin sid e)
+  [SMTPat (state_was_set tr1 prin sid e); SMTPat (tr1 <$ tr2)]
+let state_was_set_grows #a #ev tr1 tr2 prin sid e =
+  reveal_opaque (`%state_was_set) (state_was_set #a)
```
Comment (TWal) [inline src/lib/state/DY.Lib.State.Typed.fst:168]:
```suggestion
val state_was_set_grows:
  #a:Type -> {|local_state a|} ->
  tr1:trace -> tr2:trace ->
  prin:principal -> sid:state_id -> content:a  ->
  Lemma
  (requires state_was_set tr1 prin sid content /\ tr1 <$ tr2)
  (ensures state_was_set tr2 prin sid content)
  [SMTPat (state_was_set tr1 prin sid content); SMTPat (tr1 <$ tr2)]
let state_was_set_grows #a #ls tr1 tr2 prin sid content =
  reveal_opaque (`%state_was_set) (state_was_set #a)
```

File: src/lib/state/DY.Lib.State.Tagged.fst [hunk 3]
```diff
@@ -245,6 +245,16 @@ let tagged_state_was_set tr tag prin sess_id content =
...
 
+val tagged_state_was_set_grows:
+  tr1:trace -> tr2:trace ->
+  tag:string -> prin:principal -> sid:state_id -> e:bytes  ->
+  Lemma
+  (requires tagged_state_was_set tr1 tag prin sid e /\ tr1 <$ tr2)
+  (ensures tagged_state_was_set tr2 tag prin sid e)
+  [SMTPat (tagged_state_was_set tr1 tag prin sid e); SMTPat (tr1 <$ tr2)]
+let tagged_state_was_set_grows tr1 tr2 tag prin sid e =
+  reveal_opaque (`%tagged_state_was_set) (tagged_state_was_set)
```
Comment (TWal) [inline src/lib/state/DY.Lib.State.Tagged.fst:256]:
```suggestion
val tagged_state_was_set_grows:
  tr1:trace -> tr2:trace ->
  tag:string -> prin:principal -> sid:state_id -> content:bytes  ->
  Lemma
  (requires tagged_state_was_set tr1 tag prin sid content /\ tr1 <$ tr2)
  (ensures tagged_state_was_set tr2 tag prin sid content)
  [SMTPat (tagged_state_was_set tr1 tag prin sid content); SMTPat (tr1 <$ tr2)]
let tagged_state_was_set_grows tr1 tr2 tag prin sid content =
  reveal_opaque (`%tagged_state_was_set) (tagged_state_was_set)
```

**Review by TWal — COMMENTED** [review]
Looks good, aside from some naming issues (to have consistent naming across the file)

**Review by TWal — DISMISSED** [review]

**Comment by cwaldm** [comment]
I think, for consistency, we should also have the corresponding lemma in Core, i.e., for `DY.Core.Trace.Base.state_was_set`?

**Comment by TWal** [comment]
We could, but this one can be deduced automatically from the SMT by unfolding definitions (that are not opaque) and using the lemma `get_entry_at_grows` (that has an SMT pattern).

**Review by cwaldm — DISMISSED** [review]
Looks good and helpful!
(I also added those for the Tutorial Repo some time ago but forgot to mention them here...)

**Comment by cwaldm** [comment]
I know that the core one is shown automatically,
I just thought it would be nice to have the same lemmas on all levels. 
But if you don't think so, it is not blocking.

**Comment by fabian-hk** [comment]
@cwaldm Since there is the lemma `event_triggered_grows` in `DY.Core.Trace.Base.state_was_set`, I also added `state_was_set_grows` there.

**Review by cwaldm — APPROVED** [review]

**Pull request merged**